instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go

```diff
@@ -26,10 +26,11 @@ import (
 type gRPCContextKey struct{}
 
 type gRPCContext struct {
-	inMessages  int64
-	outMessages int64
-	metricAttrs []attribute.KeyValue
-	record      bool
+	inMessages    int64
+	outMessages   int64
+	metricAttrs   []attribute.KeyValue
+	metricAttrSet attribute.Set
```

Comment:
Funny story: I spent the last couple of hours going through a profile from production, and came to this repo to suggest this exact optimisation!